### ISSUE TITLE
b3d_lint_int folder is empty missing rgb.py

### ISSUE THREAD

**Issue by schroef** [Korchy/blender-color-matching#2]
bl 2.9.1
Win 10 20H2

the b4d_lint_int folder is empty, its missing the rgb.py file and the other folders

_Side note_
Nice seeing you still work on this. I bumped into you website again after search something about the nodegroup panel. THen notched the Color Matching addon again and remember working on it. I tried looking for that post but i cant find it no more, was it removed from BlenderArtist? I remember doing some work on it and adding i believe the PMS lib. 
![v121-adjsut-gui](https://user-images.githubusercontent.com/6923008/112674578-7bbb8180-8e3c-11eb-9d86-e04e15d377e2.png)
![v121-wm-popup-gui](https://user-images.githubusercontent.com/6923008/112674581-7bbb8180-8e3c-11eb-99f7-40b22d987a2a.png)

PS i got other PMS data now so perhaps i can expand it. I need to check i can find a eay conversion since it needs to list method with info in it, doing it manually is not a fun task.

I played around with some code and javascript and made this color-book finder. I found a Japanese , i believe it was Japanse, and they had a couple pms libs. It did need lots of stripping and cleaning, but i managed to get qute some nice data.

Here's a preview to that tool. http://www.promotiespullen.com/tmp/color-books/index.html

These are the libs i got for this search tool
![found-pms-libs](https://user-images.githubusercontent.com/6923008/112674572-7bbb8180-8e3c-11eb-873b-ef325e2561ed.png)

**Comment by schroef** [comment]
Okay, you seem to use this repo as a dev version, thats why it was missing. hahah that explains

**Comment by Korchy** [comment]
Yes, this is the dev version. You can download the release version from my site https://b3d.interplanety.org/en/blender-add-on-color-matching/

**Comment by schroef** [comment]
Yeah found it already. Why do add the dev mode for this, is that easier for yourself?

PS what did you think of perhaps adding more color books?

**Comment by Korchy** [comment]
The problem with the rgb.py file is the GitHub issue - it doesn't download files included in the project as submodules. But the dev mode is very useful too - it protects my code from various aggregators.

I'm always glad to add new palettes, to do that I just need a JSON file with a palette with the following content format:
[[[RGB], [NCS/RAL/PANTON/ANY_OTHER_FORMAT, CMYK (C), CMYK (U), HTML]], [...], ...]
you can see examples in the add-on code.

**Comment by schroef** [comment]
Thats weird, it should just add folder no matter what. I have couple addons with subfolders and the py files download just fine.

Ps wasnt there also a blender thread for this adodn. I remember commenting somewhere, but i dont see those comments anymore. Or perhas it was on your website, but i see there are hardly to no comments there. Not sure where it was, i do remember being in contact with couple times earlier

>I'm always glad to add new palettes, to do that I just need a JSON file with a palette with the following content format:
[[[RGB], [NCS/RAL/PANTON/ANY_OTHER_FORMAT, CMYK (C), CMYK (U), HTML]], [...], ...]
you can see examples in the add-on code.

Yes i know the format, i even used python to strip codes and generate lists that way. I hope i can make a couple more since i got those color books i showed you in that link, they are not in the correct format yet. But i think i could use python to generate the needed codes. There is however a small adjustment sometimes when we use Python to convert colors i noticed, it some times changes values a tiny bit. 

It would be nice and simple if we have a list of HEX codes, then use Python to make a list of the needed formatting you showed. We can use function to make the color codes in other format.

>it protects my code from various aggregators.

How actually, isnt it just a simple bool which you just set to False?

**Comment by Korchy** [comment]
This is not a simple folder, this folder is a "submodule". It is much convenient to use submodules, but GitHub has some problems with downloading them to users.

Yes, some comments on my site were lost when I switched off the Disqus comment system.

**Comment by schroef** [comment]
Okay thats weird, is it perhaps because of the naming. I see gitnore files in there as well, but that should not matter.

PS i just read this tip, add .keep file   But its kinda weird on the otherhand a user should need to do this.
What does that '_all' do, i cant find anything about it. You could also try just to use .gitnore in the root-folder.

**Comment by schroef** [comment]
Ahh now i get the submodels, did a little google. Its basically a sub repo inside your repo. 
Not sure how all of that works.
https://git-scm.com/book/en/v2/Git-Tools-Submodules

I did notice they used 'url' different vs your method. But i guess this is by concept that it will ignore sub items. Would be nice if there was a method to include them

**Comment by Korchy** [comment]
Yes, the submodule is the separate repository inside the current repository. Using them gets much power.

To properly getting all of this use git and its commands. GitHub is just a repository site with its own rules.
